src/main/java/cubicchunks/world/column/CubeMap.java

```diff
@@ -208,6 +204,6 @@ private int binarySearch(int cubeY) {
 			}
 		}
 
-		return mid; // not found :(
+		return start; // not found :(
```

Comment:
Turns out I over looked the "(-insertionPoint - 1)" and thought it just returned -1
Another problem is Collections.binarySearch() requires you to search for a Cube object, and not just the y coordinate of the Cube. CubeMap.cubes() only knows the y coordinate. :/